### ISSUE TITLE
Dropbox error 400

### ISSUE THREAD

**Issue by poisonborz** [Laverna/laverna#858]
Trying to use the Android app - I followed every step as instructed, but I still receive an Error 400 (invalid redirect_uri, similarly to #749. What could be the issue?

**Comment by kenoph** [comment]
I don't know, but I worked around it by using the Dropbox API way of things.
Create an app at https://www.dropbox.com/developers/apps
And put the token in laverna.
You'll then press "OK" in laverna and it will still give you an error.
This is because you have to manually add the oauth redirect urls for security reason.

So, go to `Newly created app > OAuth 2 > Redirect URIs` and put https://laverna.github.io/static-laverna/ or whatever you need.
If you still get the error, look at the page's url.
Change it from `dropbox.com/oauth2/authorize?response_type=token&client_id=XXXXXXX&redirect_uri=http://laverna.github.io/static-laverna/#notes` to `dropbox.com/oauth2/authorize?response_type=token&client_id=XXXXXXX&redirect_uri=https://laverna.github.io/static-laverna/#notes`.

Basically, you need an HTTPS redirect_uri otherwise dropbox will complain.

**Comment by howlym** [comment]
I have the same issue in Ubuntu 16.04

**Comment by Gabbalo** [comment]
Same issue using Vivaldi Browser in Windows 7.

Link is 

> https://www.dropbox.com/oauth2/authorize?response_type=token&client_id=INSERTKEYHERE&redirect_uri=https://laverna.cc/app/?cloudStorage=dropbox&dropboxKey=INSERTKEYHERE#notes

Errormessage provided by dropbox is is 

> unknown field "dropboxKey"

I added 

> https://laverna.cc/app/

and, just in case 

> https://laverna.cc/app/?cloudStorage

 to the allowed redirection pages in the dropbox-app.
